spring-cloud-netflix-sidecar/src/main/java/org/springframework/cloud/netflix/sidecar/SidecarConfiguration.java

```diff
@@ -80,13 +97,29 @@ public EurekaInstanceConfigBean eurekaInstanceConfigBean() {
 			int port = this.sidecarProperties.getPort();
 			config.setNonSecurePort(port);
 			config.setInstanceId(getDefaultInstanceId(this.env));
-			if(StringUtils.hasText(springAppName)) {
+			if (StringUtils.hasText(springAppName)) {
 				config.setAppname(springAppName);
 				config.setVirtualHostName(springAppName);
 				config.setSecureVirtualHostName(springAppName);
 			}
-			if (StringUtils.hasText(this.hostname)) {
-				config.setHostname(this.hostname);
+			String hostname = this.sidecarProperties.getHostname();
+			String ipAddress = this.sidecarProperties.getIpAddress();
+			if (!StringUtils.hasText(hostname) && StringUtils.hasText(this.hostname)) {
+				hostname = this.hostname;
+			}
+			if (StringUtils.hasText(hostname)) {
+				if (!StringUtils.hasText(ipAddress)) {
+					try {
+						ipAddress = inetUtils.convertAddress(InetAddress.getByName(hostname)).getIpAddress();
```

Comment:
I did that because if you are registering sidecar using previous trick `eureka.instance.hostname` (in order to target application on different hostname) we get following result on Eureka server http://localhost:8761/eureka/apps

Using following configuration (tested on `Camden.SR3`)

```yml
spring:
  application:
    name: sidecar

server:
  port: 5678

sidecar:
  port: 8080

eureka:
  instance:
    hostname: google.com
```

returns

```xml
<applications>
    <versions__delta>1</versions__delta>
    <apps__hashcode>UP_1_</apps__hashcode>
    <application>
        <name>SIDECAR</name>
        <instance>
            <instanceId>127.0.0.1:sidecar:5678</instanceId>
            <hostName>google.com</hostName>
            <app>SIDECAR</app>
            <ipAddr>127.0.0.1</ipAddr>
            <status>UP</status>
            ...
        </instance>
    </application>
</applications>
```

As you can see `hostName` is correctly set but ip address does not correspond to the hostname.

Thus if you are using `eureka.instance.preferIpAddress` or other way can fail.

So if I removed the hostname resolution and client do not provide ip address via `sidecar.ipAddress` (example: `Docker` you can determine hostname but generally not the ip address that will be generated when container will be run) you assume that Eureka information will be partially wrong?